src/vSphereVirtualMachineCollector.class.inc.php

```diff
@@ -354,10 +321,23 @@ static protected function DoCollectVMIPs($aMACToNetwork, $oVirtualMachine)
 							'mac' => $oNICInfo->macAddress,
 							'network' => array_key_exists($oNICInfo->macAddress, $aMACToNetwork) ? $aMACToNetwork[$oNICInfo->macAddress] : '',
 							'subnet_mask' => $sSubnetMask,
+							'id' => $oNICInfo->deviceConfigId,
 						);
 					}
 				}
 			}
+
+			// Seems useful to set also the informations for interfaces without an IP address
+			else {
+				Utils::Log(LOG_DEBUG, "No IP address found for interface ".$oNICInfo->macAddress." on network ".(array_key_exists($oNICInfo->macAddress, $aMACToNetwork) ? $aMACToNetwork[$oNICInfo->macAddress] : ''));
+				$aNWInterfaces[] = array(
+					'ip' => '',
+					'mac' => $oNICInfo->macAddress,
+					'subnet_mask' => '',
+					'network' => array_key_exists($oNICInfo->macAddress, $aMACToNetwork) ? $aMACToNetwork[$oNICInfo->macAddress] : '',
```

Comment:
I don't think interfaces without IP will have a network?

Reply:
Nevermind, this name is confusing, as it will be used as the interface name, not and IP network..